c_src/stb_image_nif.c

```diff
@@ -0,0 +1,264 @@
+#include <erl_nif.h>
+#define STB_IMAGE_IMPLEMENTATION
+#define STB_IMAGE_WRITE_IMPLEMENTATION
+#include <stb_image.h>
+#include <stb_image_write.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+#define MAX_NAME_LENGTH 9999
+
+#include "nif_utils.h"
+
+#ifdef __GNUC__
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+#pragma GCC diagnostic ignored "-Wunused-variable"
+#pragma GCC diagnostic ignored "-Wunused-function"
+#endif
+
+static ERL_NIF_TERM pack_data(ErlNifEnv *env, unsigned char *data, int x, int y, int n, int bytes_per_channel, const char *type) {
+    if (data != NULL) {
+        ErlNifBinary result;
+        if (enif_alloc_binary(x * y * n * bytes_per_channel, &result)) {
+            memcpy(result.data, data, result.size);
+            const char *channel_types[] = {"l", "la", "rgb", "rgba"};
+            const char *channels = "unknown";
+            if (n >= 1 && n <= 4) {
+                channels = channel_types[n - 1];
+            }
+            return enif_make_tuple5(env,
+                                    enif_make_atom(env, "ok"),
+                                    enif_make_binary(env, &result),
+                                    enif_make_tuple3(env,
+                                                     enif_make_int(env, y),
+                                                     enif_make_int(env, x),
+                                                     enif_make_int(env, n)),
+                                    enif_make_atom(env, type),
+                                    enif_make_atom(env, channels));
+        } else {
+            return enif_make_tuple2(env,
+                                    enif_make_atom(env, "error"),
+                                    enif_make_string(env, "out of memory", ERL_NIF_LATIN1));
+        }
+    } else {
+        return enif_make_tuple2(env,
+                                enif_make_atom(env, "error"),
+                                enif_make_string(env, "cannot decode image", ERL_NIF_LATIN1));
+    }
+}
+
+static ERL_NIF_TERM from_file(ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[]) {
+    if (argc != 3) {
+        return error(env, "expecting 3 arguments: filename, desired_channels, type");
+    }
+    char type[MAX_NAME_LENGTH];
```

Comment:
```suggestion
    char type[4];
```

Perhaps this can be changed to `char type[4]`? because we only allow `png`, `bmp`, `tga` and `jpg`